src/plugin.ts

```diff
@@ -5,6 +5,15 @@ import { generateFile } from './main';
 import { createTypeMap } from './types';
 import CodeGeneratorRequest = google.protobuf.compiler.CodeGeneratorRequest;
 import CodeGeneratorResponse = google.protobuf.compiler.CodeGeneratorResponse;
+import { FileSpec } from 'ts-poet';
+
+// Comment block at the top of every source file, since these comments require specific
+// syntax incompatible with ts-poet, we will hard-code the string and prepend to the
+// generator output.
+const formatSourceFile = (spec: FileSpec) => 
+`// @ts-nocheck
+/* eslint-disable */
```

Comment:
Per my other comment, it seems like `ts-nocheck` should really not be here, i.e. what are the compile errors we're suppressing? I'd also generally prefer removing the `eslint-disable` and instead making the generated code as idiomatic / clean as possible, but I'm not a hard -1 on that, especially if you have specific eslint errors you're getting now that are just unrealistic to have ts-proto avoid.

Reply:
@stephenh, The reasoning behind `@ts-nocheck` is that you are not in control of the client's tsconfig settings, and thus you need to generate 100% perfect code under all conditions and configurations. That just isn't reasonable and there are a few errors already if I remove that setting:
```
generatedproto.ts:127:10 - error TS6133: 'longToNumber' is declared but its value is never read.

127 function longToNumber(long: Long) {
             ~~~~~~~~~~~~

generatedproto.ts:134:6 - error TS6196: 'DeepPartial' is declared but never used.

134 type DeepPartial<T> = {
         ~~~~~~~~~~~
```

The nocheck flag is not harmful and prevents the need for consumers either edit generated files and commit them, or to generate code into a separate directory excluded from typescript.

Reply:
In regards to eslint-disable, that really isn't a possibility to solve in code generation. 2 spaces vs 4, prettier formatting, interfaces start with "I" or interfaces can't start with "I", take your pick, you can't please everyone's lint settings ;)

Reply:
eslint I'm flexible on, as you're right, `I` prefix type stuff, that's fair...it's been awhile since I've been in an eslint-rule-heavy codebase with super-specific rules.

Formatting, imo it's easier to lean on "just run prettier after codegen", which technically the code gen library we're using, `ts-poet`, _should_ do, but doesn't in the current v2 version of ts-poet we're using (the v3 version of ts-poet does run prettier automatically on the output with whatever the downstream project's prettier config is; granted this assumes they're using prettier). 

I'm still very hesitant about `ts-nocheck`, because AFAICT it literally turns off on all type checking in the file? That just seems non-negotiable imo? I get we'll have to cater to a kind of most-strict-tsconfig/whatever output, but I'd rather try and do that, than loosing type checking on the entire file.

Let me know if my understanding of ts-nocheck is incorrect.

Fwiw the `longToNumber` / `DeepPartial` are valid issues that I'll have work on a change to conditionally output instead of being unnecessary cruft in the output file.

Which, in general in terms of "unncessary/bad cruft in the output file", I have used "wow that generated code is really ugly so just turn off all the warnings" code generators a lot before, and it seemed like a cop-out insofar as it lets the code generator off the hook/still be lazy vs. just fixing their output to not suck. So, that is my general bias/guiding principle with ts-proto, is that the output should just not suck.

I'll compromise on eslint, you're right that can get esoteric, but would like to try and avoid `ts-nocheck`.